### PR TITLE
Gandi Compute Driver improvement

### PR DESCRIPTION
**Pull request by ZuluPro**
- Added sphinx documentation with example of node creation
- Added docstring for all public methods
- Added SSH keypair management: list, get, import, delete
- Change authentification initialisation in create_node: User must choose between login/password, SSH keys or both
- Added ex_get_node and ex_get_volume
- Added tests for all commit

### CODE REVIEW

File: libcloud/compute/drivers/gandi.py [hunk 1]
```diff
@@ -214,12 +260,15 @@ def create_node(self, **kwargs):
...
+        :keyword    keypairs: IDs of keypairs or Keypairs object
+        :type       keypairs: list of ``int`` or :class:`.KeyPair`
+
         :rtype: :class:`Node`
         """
 
-        if kwargs.get('login') is None or kwargs.get('password') is None:
-            raise GandiException(
-                1020, 'login and password must be defined for node creation')
+        if not kwargs.get('login') and not kwargs.get('keypairs'):
```
Comment (Kami) [inline libcloud/compute/drivers/gandi.py:269]: We need to fix this to use `auth` argument, but that's fine for now.

File: libcloud/compute/drivers/gandi.py [hunk 2]
```diff
@@ -617,3 +759,68 @@ def ex_update_disk(self, disk, new_size=None, new_name=None):
...
+        """
+        List registered key pairs.
+
+        :return:   A list of key par objects.
+        :rtype:   ``list`` of :class:`libcloud.compute.base.KeyPair`
+        """
+        kps = self.connection.request('hosting.ssh.list').object
+        return self._to_key_pairs(kps)
+
+    def get_key_pair(self, key_id):
```
Comment (Kami) [inline libcloud/compute/drivers/gandi.py]: For consistency with the base API, the first argument should be called name.

**Comment by Kami** [comment]
There are some minor style issues (method ordering), besides that it looks good to me.

**Comment by ZuluPro** [comment]
I changed `key_id` for name. I also change the function behavior for list and filter instead of directly get object.

For next commits, how must I raise Error for an unfound KeyPair ? Is there a hierarchy of Exceptions to use ?

**Comment by Kami** [comment]
@ZuluPro Great.

There is `libcloud.compute.types.KeyPairDoesNotExistError`.

**Comment by Kami** [comment]
Changes merged into trunk. Thanks.

**Pull request closed without merging**